astropy/units/quantity.py

```diff
@@ -593,17 +593,20 @@ def __array_finalize__(self, obj):
         if obj is None or obj.__class__ is np.ndarray:
             return
 
-        # If our unit is not set and obj has a valid one, use it.
+        # Copy over the unit and possibly info.  Note that the only way the
+        # unit can already be set is if one enters via _new_view(), where the
+        # unit is often different from that of self, and where propagation of
+        # info is not always desirable.
         if self._unit is None:
             unit = getattr(obj, "_unit", None)
             if unit is not None:
                 self._set_unit(unit)
 
-        # Copy info if the original had `info` defined.  Because of the way the
-        # DataInfo works, `'info' in obj.__dict__` is False until the
-        # `info` attribute is accessed or set.
-        if "info" in obj.__dict__:
-            self.info = obj.info
+            # Copy info if the original had `info` defined.  Because of the way the
+            # DataInfo works, `'info' in obj.__dict__` is False until the
+            # `info` attribute is accessed or set.
+            if "info" in obj.__dict__:
```

Comment:
Moving the info into the unit check makes a lot of sense for `Quantity`, but what about, e.g. Longitude, which has a wrap-angle attribute. Should a similar `info` check be done in this case?

Reply:
`Longitude` sets wrap_angle in `__array_finalize__` so that will still work. I also `super()`'s, so `info` gets propagated for it here.

Or do you mean whether the copying of `wrap_angle` should be moved inside an `if self._unit is None` stanza? I don't think so since `wrap_angle` always needs to be set (unlike `info`).